src/qarchivecompressor_p.cc

```diff
@@ -234,69 +234,35 @@ void CompressorPrivate::addFiles(const QStringList &entryNames, const QStringLis
 }
 
 void CompressorPrivate::removeFiles(const QString &entry) {
-    if(b_Started || b_Paused) {
+    if(b_Started || b_Paused)
         return;
-    }
-    int index = 0;
-    for(const auto& file : m_StaggedFiles) {
-        if(file && file->entry == entry) {
-            m_StaggedFiles.remove(index);
-            return;
-        }
-        ++index;
-    }
+
+    auto it = std::find_if(m_StaggedFiles.begin(), m_StaggedFiles.end(), [&](const Node* f){ return f && f->entry == entry; });
+    if (it != m_StaggedFiles.end())
+        m_StaggedFiles.remove(std::distance(m_StaggedFiles.begin(), it));
 }
 
 void CompressorPrivate::removeFiles(const QStringList &entries) {
-    if(b_Started || b_Paused) {
+    if(b_Started || b_Paused)
         return;
-    }
+
     QVector<int> indexes;
-    int index = 0;
-    for(const auto& file : m_StaggedFiles) {
-        if(file && entries.contains(file->entry)) {
-            indexes.append(index);
-            return;
-        }
-        ++index;
+    auto it = std::find_if(m_StaggedFiles.begin(), m_StaggedFiles.end(), [&](const Node* f) { return f && entries.contains(f->entry); });
+    if (it != m_StaggedFiles.end()) {
+        indexes.append(std::distance(m_StaggedFiles.begin(), it));
+	return;
```

Comment:
This might have been a error on my side, if the code returns here then nothing will be removed. My bad... We should remove this return statement here. I think a major refactor is long overdue. :octocat:

Reply:
Also make sure to rebase and force push since there is a lot of changes to master.

Reply:
Rebased.